test/e2e/common/events.go

```diff
@@ -33,8 +34,89 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 
 	"github.com/onsi/ginkgo"
+	"k8s.io/apimachinery/pkg/types"
 )
 
+var _ = ginkgo.Describe("[sig-api-machinery] Events", func() {
+	f := framework.NewDefaultFramework("events")
+
+	ginkgo.It("should ensure that an event can be fetched, patched, deleted, and listed", func() {
+		eventTestName := "event-test"
+
+		ginkgo.By("creating a test event")
+		// create a test event in test namespace
+		_, err := f.ClientSet.CoreV1().Events(f.Namespace.Name).Create(&v1.Event{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: eventTestName,
+				Labels: map[string]string{
+					"testevent-constant": "true",
+				},
+			},
+			Message: "This is a test event",
+			Reason:  "Test",
+			Type:    "Normal",
+			Count:   1,
+			InvolvedObject: v1.ObjectReference{
+				Namespace: f.Namespace.Name,
+			},
+		})
+		framework.ExpectNoError(err, "failed to create test event")
+
+		ginkgo.By("listing all events in all namespaces")
+		// get a list of Events in all namespaces
+		eventsList, err := f.ClientSet.CoreV1().Events("").List(metav1.ListOptions{
+			LabelSelector: "testevent-constant=true",
+		})
+		framework.ExpectNoError(err, "failed list all events")
+
+		foundCreatedEvent := false
+		var eventCreatedName string
+		for _, val := range eventsList.Items {
+			if val.ObjectMeta.Name == eventTestName && val.Message == "This is a test event" {
```

Comment:
look for it by name/namespace not name/data

Reply:
Updated in 2a535e4f38c